src/mlpack/bindings/markdown/print_docs.cpp

```diff
@@ -40,10 +42,13 @@ void PrintHeaders(const std::string& bindingName,
 
     // Get the name of the binding in the target language, and convert it to
     // lowercase (since the anchor link will be in lowercase).
-    const std::string langBindingName = GetBindingName(bindingName);
-    std::string anchorName = langBindingName;
-    std::transform(anchorName.begin(), anchorName.end(), anchorName.begin(),
-        [](unsigned char c) { return std::tolower(c); });
+    const string langBindingName = 
+        addWrapperDocs[i] ? GetWrapperName(bindingName) :
+            GetBindingName(bindingName);
+
+    string anchorName = GetBindingName(bindingName);
+    transform(anchorName.begin(), anchorName.end(), anchorName.begin(),
+        [](unsigned char c) { return tolower(c); });
```

Comment:
Yes, that is correct, GetBindingName does not give the "class" name, I just created a new function GetWrapperName because GetBindingName will still be used for bindings that are not yet refactored.

Reply:
However, there is still some issue with the link, I will change that too.

Reply:
I changed the link of the sidebar to class-adaboost. In mlpack.md it looks like this:

```markdown
<div class="language-title" id="python" markdown="1">
## class Adaboost
{: #python_class-adaboost }
</div>
```

shouldn't this be linking [here](http://mlpack2.ratml.org/doc/mlpack-git/python_documentation.html#class-adaboost) ?
Let me know if I am missing something?

Reply:
maybe we can get away with this by just changing `class Adaboost()` to `Adaboost()` ?

Reply:
I'm not sure I understand fully.  When I take a look at `mlpack.md` as generated in commit 9d4d4ba7, I see the `language-title` block just like you posted, but the sidebar code is still pointing at the wrong anchor:

```
 - [class Adaboost](#python_adaboost){: .language-link #python }
```

It seems to me like the issue will be fixed if we can emit `#python_class-adaboost` for that link.

But, maybe I have misunderstood a deeper point here?